File: Assesments/Vetrivel_Assesment-SeleniumIde-pom/Technical-assesmnet/src/test/java/com/pages/SignupPage.java
package com.pages;

import java.time.Duration;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.Select;
import org.openqa.selenium.support.ui.WebDriverWait;

public class SignupPage extends BasePage {

    private WebDriverWait wait;

    public SignupPage(WebDriver driver) {
        super(driver);
        wait = new WebDriverWait(driver, Duration.ofSeconds(10));
    }

    By signupPassword=By.id("password");
    By firstName=By.id("first_name");
    By lastName=By.id("last_name");
    By phone=By.id("mobile_number");
    By address=By.id("address1");
    By countryDropDown=By.id("country");
    By state=By.id("state");
    By city=By.id("city");
    By zipCode=By.id("zipcode");
    
    By createAccount=By.xpath("//*[@id=\"form\"]/div/div/div/div/form/button");

    private WebElement waitForElement(By locator) {
        return wait.until(ExpectedConditions.visibilityOfElementLocated(locator));
    }

    public void enterPassword(String password) {
        waitForElement(signupPassword)
        .sendKeys(password);
    }

    public void enterFirstName(String fName) {
        waitForElement(firstName)
        .sendKeys(fName);
    }

    public void enterLastName(String lName) {
        waitForElement(lastName)
        .sendKeys(lName);
    }

    public void enterPhone(String mobile) {
        waitForElement(phone)
        .sendKeys(mobile);
    }

    public void enterAddress(String addr) {
        waitForElement(address)
        .sendKeys(addr);
    }

    public void selectCountry(String countryName) {
        WebElement dropdown = wait.until(ExpectedConditions.elementToBeClickable(countryDropDown));
        Select select=new Select(dropdown);
        select.selectByVisibleText(countryName);
    }

    public void enterState(String stateName) {
        waitForElement(state)
        .sendKeys(stateName);
    }

    public void enterCity(String cityName) {
        waitForElement(city)
        .sendKeys(cityName);
    }

    public void enterZipCode(String zip) {
        waitForElement(zipCode)
        .sendKeys(zip);
    }
    
    public void clickCreateAccount() {
    	wait.until(ExpectedConditions.elementToBeClickable(createAccount))
    	.click();
    }
}

File: TypeScript/PlayWright/AutomationTestcasesPOM/pages/signupPage.ts
import {Page} from "@playwright/test"
import {BasePage} from "../pages/basePage"

export class SignupPage extends BasePage{

    readonly name;
    readonly email;
    readonly submit;

    readonly password;

    readonly firstName;
    readonly lastName;
    readonly address;
    readonly state;
    readonly city;
    readonly zipcode;
    readonly mobile;
    readonly createAccount;

    readonly accountCreatedMessage;

    constructor(page : Page){

        super(page)
        this.name=page.locator("//input[@placeholder='Name']")
        this.email=page.locator("//input[@data-qa='signup-email']")
        this.submit=page.locator("//button[normalize-space()='Signup']")

        this.password=page.locator("//input[@id='password']")

        this.firstName=page.locator("//input[@id='first_name']")
        this.lastName=page.locator("//input[@id='last_name']")
        this.address=page.locator("//input[@id='address1']")
        this.state=page.locator("//input[@id='state']")
        this.city=page.locator("//input[@id='city']")
        this.zipcode=page.locator("//input[@id='zipcode']")
        this.mobile=page.locator("//input[@id='mobile_number']")

        this.createAccount=page.locator("//button[normalize-space()='Create Account']")

        this.accountCreatedMessage=page.locator("//b[normalize-space()='Account Created!']")

    }

    async fillDetails(name : string, email : string){
        await this.name.fill(name)
        await this.email.fill(email)
        await this.submit.click()
    }

    async signUp(password : string, firstName : string, lastName : string, address : string, state : string, city : string, zipcode : string, mobile : string){
        
        await this.password.fill(password)
        await this.firstName.fill(firstName)
        await this.lastName.fill(lastName)
        await this.address.fill(address)
        await this.state.fill(state)
        await this.city.fill(city)
        await this.zipcode.fill(zipcode)
        await this.mobile.fill(mobile)

        await this.createAccount.click()
    }

    async isAccountCreated(){
        return await this.accountCreatedMessage.isVisible()
    }




}